### PR TITLE
Lesson 7: Character driver

### PR DESCRIPTION
**Pull request by Noxybot**

### CODE REVIEW

File: 02-KernelModules_CharacterDevice/character_driver.c [hunk 1]
```diff
@@ -0,0 +1,242 @@
...
+static void delete_buffer(void){
+    if(driver_buffer != NULL){
+        kfree(driver_buffer);
+        driver_buffer = NULL;
+    }
+    data_size = 0;
+}
+
+//DEVICE CREATION
+#define NUMBER_OF_DEVICES   1
```
Comment (oleg-h2o) [inline 02-KernelModules_CharacterDevice/character_driver.c]: Plz move all #define(s) to the upper part of this file.

File: 02-KernelModules_CharacterDevice/character_driver.c [hunk 2]
```diff
@@ -0,0 +1,242 @@
...
+    if(IS_ERR(my_class))
+        return -1;
+    device_create(my_class, NULL, dev, NULL, DEVICE_NAME );
+    return 0;
+    printk(KERN_DEBUG "Add dev %d -- %d\n", MAJOR(dev), MINOR(dev));
+}
+
+
+// /proc INTERFACE
+#define PROC_DIR_NAME    "My_proc_stats"
```
Comment (oleg-h2o) [inline 02-KernelModules_CharacterDevice/character_driver.c]: .. and these ones also...

File: 02-KernelModules_CharacterDevice/character_driver.c [hunk 3]
```diff
@@ -0,0 +1,242 @@
...
+
+static void init_proc(void)
+{
+    proc_dir = proc_mkdir(PROC_DIR_NAME, NULL);
+    proc_info_file = proc_create(PROC_FILE_NAME, S_IRUGO, proc_dir, &proc_ops);
+}
+
+
+// /sys INTERFACE
+#define SYS_DIR_NAME    "My_sys_operations"
```
Comment (oleg-h2o) [inline 02-KernelModules_CharacterDevice/character_driver.c]: ... and these.

File: 02-KernelModules_CharacterDevice/Makefile [hunk 4]
```diff
@@ -0,0 +1,12 @@
+KDIR := /home/eduard/Kernel_i386
```
Comment (galkinletter) [inline 02-KernelModules_CharacterDevice/Makefile]:
```suggestion
KDIR := /home/eduard/Kernel_i386
```
$KDIR refers to the path of the kernel source directory

Please use variable  as path

No hardcoded paths!

**Review by galkinletter — CHANGES_REQUESTED** [review]
Need fix

**Comment by galkinletter** [comment]
No hardcoded paths! Please fix also

**Comment by galkinletter** [comment]
Ok

**Pull request merged**